primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java

```diff
@@ -204,15 +215,17 @@ protected <F extends Comparable> Predicate createPredicate(FilterMeta filter, Fi
             case NOT_EQUALS:
                 return cb.notEqual(fieldExpression, filterValue);
             case LESS_THAN:
-                return cb.lessThan(fieldExpression, filterValue);
+                return cb.lessThan(fieldExpression, (Comparable) filterValue);
             case LESS_THAN_EQUALS:
-                return cb.lessThanOrEqualTo(fieldExpression, filterValue);
+                return cb.lessThanOrEqualTo(fieldExpression, (Comparable) filterValue);
             case GREATER_THAN:
-                return cb.greaterThan(fieldExpression, filterValue);
+                return cb.greaterThan(fieldExpression, (Comparable) filterValue);
             case GREATER_THAN_EQUALS:
-                return cb.greaterThanOrEqualTo(fieldExpression, filterValue);
+                return cb.greaterThanOrEqualTo(fieldExpression, (Comparable) filterValue);
             case IN:
-                throw new UnsupportedOperationException("MatchMode.IN currently not supported!");
+                return filterValueAsCollection.get().size() == 1
+                        ? cb.equal(fieldExpression, filterValueAsCollection.get().iterator().next())
+                        : fieldExpression.in(filterValueAsCollection.get());
             case NOT_IN:
                 throw new UnsupportedOperationException("MatchMode.NOT_IN currently not supported!");
```

Comment:
You should be able to add Not In too right?

Reply:
I don't think there's a straight-forward `notIn` method, so I need to refresh myself on how to negate predicates in Criteria API. I wanted to get this in first before I go digging to see how we do it. But, yeah, it should be simple to tack in.

Reply:
... and [it was](https://github.com/primefaces/primefaces/pull/9325/commits/282310e1aece020c7898e5dc6a183f2cb8c1e344)